bitcoin_client_rs/tests/client.rs

```diff
@@ -310,15 +310,50 @@ async fn test_sign_psbt() {
             .map(|v| serde_json::from_value(v.clone()).unwrap())
             .unwrap();
 
-        let psbt: Psbt = deserialize(&base64::decode(&psbt_str).unwrap()).unwrap();
+        let psbt = Psbt::deserialize(&base64::decode(&psbt_str).unwrap()).unwrap();
 
         let wallet = wallet::WalletPolicy::new(name, wallet::Version::V2, policy, keys);
 
         let store = utils::RecordStore::new(&exchanges);
-        let _res = client::BitcoinClient::new(utils::TransportReplayer::new(store.clone()))
+        let res = client::BitcoinClient::new(utils::TransportReplayer::new(store.clone()))
             .sign_psbt(&psbt, &wallet, hmac.as_ref())
             .unwrap();
 
+        for (i, psbt_sig) in res {
+            for (j, res_sig) in sigs.iter().enumerate() {
+                if i == j {
+                    match psbt_sig {
+                        PartialSignature::TapScriptSig(key, tapleaf_hash, sig) => {
+                            assert_eq!(
+                                res_sig
+                                    .get("key")
+                                    .map(|v| serde_json::from_value::<String>(v.clone()).unwrap())
+                                    .unwrap(),
+                                key.to_string()
+                            );
+                            if let Some(tapleaf_hash_res) = res_sig
+                                .get("tapleaf_hash")
+                                .map(|v| serde_json::from_value::<String>(v.clone()).unwrap())
+                            {
+                                assert_eq!(
+                                    tapleaf_hash_res,
+                                    hex::encode(tapleaf_hash.unwrap().to_byte_array())
+                                );
+                            }
+                            assert_eq!(
+                                res_sig
+                                    .get("sig")
+                                    .map(|v| serde_json::from_value::<String>(v.clone()).unwrap())
+                                    .unwrap(),
+                                hex::encode(sig.to_vec())
+                            );
+                        }
+                        _ => {}
+                    }
+                }
+            }
+        }
+
```

Comment:
There's identical code in lines 362:395, perhaps it can be in a function?
Even an inner function might do.